File: Assemblies/Eonic.Tools/Integration/Twitter/TwitterVB2/Objects/ATwiCliVideoFile.vb
Namespace Integration.Twitter.TwitterVB2
    Public Class ATwiCliVideoFile
        Inherits ATwiCliFile
#Region "TCVF Constants"
        Public Const TCVF_VIDEO_STRING As String = "video.flv"
        Public Const TCVF_VIDEO_EXTENSION As String = ".flv"
#End Region
        Private m_strThumbNailURL As String

        Public Property ThumbNailURL() As String
            Get
                Return m_strThumbNailURL

            End Get
            Set(ByVal value As String)
                m_strThumbNailURL = value
            End Set
        End Property


        Public Overrides Function GetDirectURL() As String
            Return TCF_DIRECT_URL & TCVF_VIDEO_STRING & "?id=" & Me.ID & "&size=original"
        End Function

        Public Overrides Function GetExtension() As String
            Return TCVF_VIDEO_EXTENSION

        End Function
    End Class

End Namespace

File: Assemblies/Protean.Tools/Integration/Twitter/TwitterVB2/Objects/ATwiCliVideoFile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Protean.Tools.Integration.Twitter.TwitterVB2
{
    public partial class ATwiCliVideoFile : ATwiCliFile
    {
        #region TCVF Constants
        public const string TCVF_VIDEO_STRING = "video.flv";
        public const string TCVF_VIDEO_EXTENSION = ".flv";
        #endregion
        private string m_strThumbNailURL;

        public string ThumbNailURL
        {
            get
            {
                return m_strThumbNailURL;

            }
            set
            {
                m_strThumbNailURL = value;
            }
        }


        public override string GetDirectURL()
        {
            return TCF_DIRECT_URL + TCVF_VIDEO_STRING + "?id=" + this.ID + "&size=original";
        }

        public override string GetExtension()
        {
            return TCVF_VIDEO_EXTENSION;

        }
    }

}
